scripts/core/exp-command-buffer.yml

```diff
@@ -217,6 +217,51 @@ returns:
     - $X_RESULT_ERROR_OUT_OF_RESOURCES
 --- #--------------------------------------------------------------------------
 type: function
+desc: "Append a USM fill command to a command-buffer object"
+class: $xCommandBuffer
+name: AppendFillUSMExp
+params:
+    - type: $x_exp_command_buffer_handle_t
+      name: hCommandBuffer
+      desc: "[in] handle of the command-buffer object."
+    - type: "void*"
+      name: pMemory
+      desc: "[in] pointer to USM memory object."
+    - type: "const void*"
+      name: pPattern
+      desc: "[in] pointer to the fill pattern."
+    - type: "size_t"
+      name: patternSize
+      desc: "[in] size in bytes of the pattern."
```

Comment:
Are we expecting any restrictions on `patternSize`? e.g Must be a power-of-two, or <= the largest integer or floating-point vector data type supported by the device.

Same comment for buffer equivalent entry-point

Reply:
There are restrictions listed in the `returns:` section that are the same as the existing APIs:

```rst
    - $X_RESULT_ERROR_INVALID_SIZE:
        - "`patternSize == 0 || size == 0`"
        - "`patternSize > size`"
        - "`(patternSize & (patternSize - 1)) != 0`"
        - "`size % patternSize != 0`"
        - "If `size` is higher than the allocation size of `ptr`"
```

Reply:
apologies, missed that. Looks good.

Because `size` has ",must be a multiple of patternSize." after its description it got me thinking down this path.